main.go

```diff
@@ -325,6 +328,24 @@ func main() { //nolint:funlen,maintidx
 		os.Exit(1)
 	}
 
+	ons, err := cluster.GetOperatorNamespace()
+	if err != nil {
+		setupLog.Error(err, "unable to determine Operator Namespace")
+		os.Exit(1)
+	}
+
+	gc.Instance = gc.New(
+		oc,
+		ons,
+		gc.WithUnremovables(gvk.CustomResourceDefinition, gvk.Lease),
```

Comment:
not understand here: 
without state here, it could delete Lease instance?

Reply:
the gc service uses the `SelfSubjectRulesReview` to retrieve the list of resources it can delete, given the role & role binding configured for the operator, hence it could delete `CRDs` and `Leases`. This is mainly to add an additional gate to avoid it could happen by mistake.